ciao-controller/internal/datastore/datastore.go

```diff
@@ -401,26 +401,24 @@ func newHardwareAddr() (net.HardwareAddr, error) {
 func (ds *Datastore) AddTenant(id string) (*types.Tenant, error) {
 	hw, err := newHardwareAddr()
 	if err != nil {
-		glog.V(2).Info("error creating mac address", err)
-		return nil, err
+		return nil, errors.Wrap(err, "error creating MAC address")
 	}
 
 	err = ds.db.addTenant(id, hw.String())
 	if err != nil {
-		glog.V(2).Info("error adding tenant", err)
-		return nil, err
+		return nil, errors.Wrapf(err, "error adding tenant (%v) to database", id)
 	}
 
 	t, err := ds.getTenant(id)
 	if err != nil || t == nil {
-		glog.V(2).Info(err, " unable to get tenant: ", id)
+		return nil, err
```

Comment:
Yes intentionally base. As ds.getTenant() wraps and we can't add any useful context over that one.